src/main/scala/org/apache/spark/sql/redis/HashRedisPersistence.scala

```diff
@@ -30,28 +34,35 @@ class HashRedisPersistence extends RedisPersistence[Map[String, String]] {
     }
   }
 
-  override def encodeRow(value: Row): Map[String, String] = {
+  override def encodeRow(key: (String, String), value: Row): Map[String, String] = {
     val fields = value.schema.fields.map(_.name)
     val kvMap = value.getValuesMap[Any](fields)
     kvMap
-      .filter { case (k, v) =>
+      .filter { case (_, v) =>
         // don't store null values
         v != null
       }
+      .filter { case (k, _) =>
+        // don't store key values
+        k != key._1
+      }
       .map { case (k, v) =>
         k -> String.valueOf(v)
       }
   }
 
-  override def decodeRow(value: Map[String, String], schema: => StructType,
-                         inferSchema: Boolean): Row = {
+  override def decodeRow(key: (String, String), value: Any, schema: => StructType,
+                         inferSchema: Boolean, requiredColumns: Seq[String]): Row = {
+    val values = value match {
+      case v: JMap[String, String] => v.asScala.toSeq
```

Comment:
There are compilation warning related to type erasure. Can we fix them?

Reply:
I think we can make an explicit cast. That will not make the match safer, but at least the warnings will go aways